libs/platform/src/lib/components/form/text-area/text-area.component.ts

```diff
@@ -0,0 +1,395 @@
+            const textareaTotalHeight = this._getTextareaTotalHeight();
+            const maxHeight = parseInt(this.textareaElement.nativeElement.style.maxHeight as string, 10);
+            if (maxHeight) {
+                // console.log('total height' + textareaTotalHeight);
+                if (textareaTotalHeight <= maxHeight) {
+                    this.textareaElement.nativeElement.style.height = textareaTotalHeight + 'px';
+                } else {
+                    if (this.height) {
+                        this.textareaElement.nativeElement.style.height = this.height;
+                    } else {
+                        this.textareaElement.nativeElement.style.height = maxHeight + 'px';
+                    }
+                }
+            } else {
+                // no bound max height, keep growing
+                this.textareaElement.nativeElement.style.height = textareaTotalHeight + 'px';
+            }
+        }
+    }
+
+    /**
+     * get line height of textarea
+     */
+    getTextareaLineHeight(): number {
+        // Get the computed styles for the element
+        if (this.textareaElement && this.textareaElement.nativeElement) {
+            const computed = window.getComputedStyle(this.textareaElement.nativeElement);
+
+            // Calculate the height
+            return parseInt(computed.getPropertyValue('line-height'), 10);
+        }
+        return 20; // default line height if nothing is defined
+    }
+
+    /** when backpress or delete key is pressed when showExceededText field is not set, simply remove excess characters,
+    /* else handle autogrow case
+    */
+    @HostListener('keyup', ['$event'])
+    handleBackPress(event: KeyboardEvent) {
+        // if not showing exceeded text message/interactions or growing, and custom value set
+        if (this.growing) {
+            this.autoGrowTextArea();
+            this.isValueCustomSet = false; // set it to false first time it comes here
+        }
+        if (!this.showExceededText && (event.keyCode === DELETE || event.keyCode === BACKSPACE)) {
+            // for the custom value set and showExceededText=false case, on any key press, remove excess characters
+            if (this.value) {
+                this._textAreaCharCount = this.value.length;
+                if (this._textAreaCharCount > this.maxLength) {
+                    // remove excess characters
+                    this.value = this.value.substring(0, this.maxLength);
+                    this.isValueCustomSet = false; // since value is now changed, it is no longer custom set
+                }
+            }
+        }
+    }
+
+    /**
+     * get the updated state when character count breaches `maxLength`
+     */
+    getUpdatedState(): string {
+        if (
+            ((this.textareaElement && this.textareaElement.nativeElement.value.length) ||
+                (this.value && this.value.length)) > this.maxLength
```

Comment:
I am not sure if I understand this fragment 
`(this.value && this.value.lenght)` > this.maxLenght. 

The blog where you expect a value is boolean expression

Reply:
This block checks whether the underlying element ref `textareaElement` or the `value` field, if it has some content, if the length is greater than the `maxLength` value.
The reason we are looking at both `textareaElement` and the `value` fields is because if value is initially set through `fdp-form-field`, then `textareaElement` will be undefined, therefore we are checking for `value` as well.

Reply:
Can you restructure this is not good for readability one thing is to test if value is defined and thing is to compare length don't mix those two

(this.value && this.value.lenght > xxxx) instead of (this.value && this.value.length)) > this.maxLength

Reply:
Okay I see the readability issue, the second bracket was just an extra wrapping. I will move this condition to a private method and refer to it like `if (getContentLength() > maxLength)` instead.

Reply:
have restructured it now